File: frontend/classes/Fleet.ts
export interface FleetData {
  id: number
  game: number
  number: number
  name: string
  campaign: number | null
}

class Fleet {
  id: number
  game: number
  number: number
  name: string
  campaign: number | null

  constructor(data: FleetData) {
    this.id = data.id
    this.game = data.game
    this.number = data.number
    this.name = data.name
    this.campaign = data.campaign
  }
}

export default Fleet

File: backend/rorapp/models/fleet.py
import roman
from django.db import models
from django.core.validators import MinValueValidator, MaxValueValidator

from rorapp.models.game import Game


class Fleet(models.Model):
    game = models.ForeignKey(Game, related_name="fleets", on_delete=models.CASCADE)
    number = models.IntegerField(
        validators=[MinValueValidator(1), MaxValueValidator(25)]
    )

    @property
    def name(self):
        return roman.toRoman(self.number)

    class Meta:
        constraints = [
            models.UniqueConstraint(
                fields=["game", "number"], name="unique_fleet_game_number"
            )
        ]
